### PR TITLE
Image Alt Text Implemented

### PR DESCRIPTION
**Pull request by BeanSampa**
Hello @alonrbar, due to project requirements I implemented ALT TEXT for the images. It's just a few tweaks. I took care that if the "descr" parameter is sent blank, it defines the image as "decorative", this is important, it will not generate an error in Word (if the Accessibility Check is enabled) and it makes the conversion of the PDF document more accurate.
This is my first contribution to a project, so if I made a mistake, I apologize.
Thanks for sharing your code, it's been a big help.

### CODE REVIEW

File: src/plugins/image/imagePlugin.ts [hunk 1]
```diff
@@ -95,7 +103,13 @@ export class ImagePlugin extends TemplatePlugin {
         return `
             <pic:pic xmlns:pic="http://schemas.openxmlformats.org/drawingml/2006/picture">
                 <pic:nvPicPr>
-                    <pic:cNvPr id="0" name="${name}"/>
+                    <pic:cNvPr id="${imageId}" name="${name}" ${descrTag}>
```
Comment (alonrbar) [inline src/plugins/image/imagePlugin.ts:106]: Are you sure this part is required? I don't see it when I add a picture on my machine.

**Review by alonrbar — COMMENTED** [review]
Hi, thank you for your contribution!
Overall looks great, but please address the comments below.

File: src/plugins/image/imagePlugin.ts (hunk 1 above)
Comment (BeanSampa) [reply src/plugins/image/imagePlugin.ts:106]: I created a word document and extracted the ZIP, and got the below.
Microsoft Office 365 - Word Version 2302 (build 16130.20332)
```
<wp:docPr id="1" name="Image 1" descr="This is AltText"/>
<wp:cNvGraphicFramePr>
<a:graphicFrameLocks xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main" noChangeAspect="1"/>
</wp:cNvGraphicFramePr>
<a:graphic xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main">
<a:graphicData uri="http://schemas.openxmlformats.org/drawingml/2006/picture">
<pic:pic xmlns:pic="http://schemas.openxmlformats.org/drawingml/2006/picture">
<pic:nvPicPr>
<pic:cNvPr id="1" name="Image 1" descr="This is AltText"/>
<pic:cNvPicPr/>
</pic:nvPicPr>
```

File: README.md [hunk 2]
```diff
@@ -268,6 +268,7 @@ Input data:
     "Kung Fu Hero": {
         _type: "image",
         source: fs.readFileSync("hero.png"),
+        descr: "Kung Fu Hero" // This is the Image's Alt Text - If it equals "", the image will be decorative (no Alt Text)
```
[the remark this answers is not in the document]
Comment (BeanSampa) [reply README.md]: Done

File: src/plugins/image/imageContent.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ export type ImageFormat = MimeType.Jpeg | MimeType.Png | MimeType.Gif | MimeType
 export interface ImageContent extends PluginContent {
     _type: 'image';
     source: Binary;
+    descr: string;
```
[the remark this answers is not in the document]
Comment (BeanSampa) [reply src/plugins/image/imageContent.ts]: Done

File: src/plugins/image/imagePlugin.ts [hunk 4]
```diff
@@ -59,18 +59,26 @@ export class ImagePlugin extends TemplatePlugin {
         //
 
         const name = `Picture ${imageId}`;
+        const decorative = descr==='' ? 1 : 0;
+        const descrTag = decorative===0 ? `descr="${descr}"` : '';
```
[the remark this answers is not in the document]
Comment (BeanSampa) [reply src/plugins/image/imagePlugin.ts]: Done

**Comment by BeanSampa** [comment]
Fixed the tests files (/test/fixture/): header.tests.ts and image.tests.ts

**Comment by alonrbar** [comment]
I see it still fails in validation but never mind. I'll merge, fix it on my machine and upload a fix before releasing an new version of the library. Thanks!

**Comment by alonrbar** [comment]
Alt text is now part of v3.1.0

**Pull request merged**